contracts/proxy/utils/UUPSUpgradeable.sol

```diff
@@ -34,6 +35,23 @@ abstract contract UUPSUpgradeable is ERC1967Upgrade {
         _;
     }
 
+    /**
+     * @dev Check that the execution is not being performed through a delegate call. This allows a function to be
+     * callable on the implementing contract but not through proxies.
+     */
+    modifier notDelegated() {
```

Comment:
`onlyProxy` requires the execution context to have delegate to this contract, and require the execution context to be that of a proxy pointing to self.... hence the name "onlyProxy".

`notDelegated` only accepts execution context that are not delegated. This will revert if called by a proxy (minimal, uups, transparent, beacon, ... whatever). It is not exactly the opposite of onlyProxy, but its close, and there is 0 overlap.

There are 3 possible options:
1. call was not delegated
    - `onlyProxy` → reverts
    - `notDelegated` → ok
2. call was delegated by a ERC1967 proxy (uups or transparent)
    - `onlyProxy` → ok
    - `notDelegated` → reverts
3. call was delegated by another mechanism (minimal clone, beacon proxy, ...)
    - `onlyProxy` → reverts
    - `notDelegated` → reverts
    
It the current natspec comment not enough?

Reply:
Yeah sorry I thought the only difference was in checking for `_getImplementation`.